numpy/core/fromnumeric.py

```diff
@@ -445,6 +445,10 @@ def put(a, ind, v, mode='raise'):
     array([ 0,  1,  2,  3, -5])
 
     """
+    if not isinstance(a, np.ndarray):
```

Comment:
Hmmm...I'm a little on the fence on this one.  While I do agree it should be faster in the most common case, coupling the type checking with the function call makes us dependent on an ```AttributeError``` when calling the C API should ```a``` not be an ```ndarray```.  Should the API change, say to something like what is done with ```np.place```, that ```try...except``` would become moot.